packages/venia-ui/lib/components/ProductOptions/option.js

```diff
@@ -1,62 +1,87 @@
-import React, { Component } from 'react';
+import React, { useCallback, useMemo } from 'react';
 import { arrayOf, func, object, shape, string } from 'prop-types';
 
-import classify from '../../classify';
+import { mergeClasses } from '../../classify';
 import getOptionType from './getOptionType';
 import SwatchList from './swatchList';
 import TileList from './tileList';
 import defaultClasses from './option.css';
 
-const getItemKey = ({ value_index }) => value_index;
+const Option = props => {
+    const {
+        attribute_code,
+        attribute_id,
+        label,
+        onSelectionChange,
+        selectedVariant,
+        values
+    } = props;
 
-class Option extends Component {
-    static propTypes = {
-        attribute_id: string,
-        attribute_code: string.isRequired,
-        classes: shape({
-            root: string,
-            title: string
-        }),
-        label: string.isRequired,
-        onSelectionChange: func,
-        values: arrayOf(object).isRequired
-    };
+    const classes = mergeClasses(defaultClasses, props.classes);
 
-    handleSelectionChange = selection => {
-        const { attribute_id, onSelectionChange } = this.props;
-
-        if (onSelectionChange) {
-            onSelectionChange(attribute_id, selection);
-        }
-    };
-
-    get listComponent() {
-        const { attribute_code, values } = this.props;
+    const handleSelectionChange = useCallback(
+        selection => {
+            if (onSelectionChange) {
+                onSelectionChange(attribute_id, selection);
+            }
+        },
+        [attribute_id, onSelectionChange]
+    );
 
+    const CustomList = useMemo(() => {
         // TODO: get an explicit field from the API
         // that identifies an attribute as a swatch
         const optionType = getOptionType({ attribute_code, values });
 
         return optionType === 'swatch' ? SwatchList : TileList;
-    }
+    }, [attribute_code, values]);
+
+    let initialSelection = {};
+    if (selectedVariant) {
+        // If we have a selected variant, pass it through to List.
+        // Attempt to find an option of the variant that's one of the product's configurable options.
+        // If found, set that configurable option as the initial selection of the List.
+        selectedVariant.options.find(option => {
```

Comment:
We can use the new `productVariants` utility for this instead.

Reply:
Turns out we don't want to do this: we want this match to be restricted only to the `values` that this list is passed (as opposed to ALL `configurable_options` values).

Reply:
We may be able to add a function to the `productVariants` utility for this ^ use case.

Reply:
Eh, we'll add it if/when we need it later.